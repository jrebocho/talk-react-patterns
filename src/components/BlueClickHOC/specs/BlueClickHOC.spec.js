import React from 'react'
import { shallow } from 'enzyme'
import withBlueClickHOC from '../../BlueClickHOC'

describe('.withBlueClickHOC', () => {
  it('renders correctly', () => {
    const TestComponent = () => <div>test</div>
    const BlueClickComponent = withBlueClickHOC(TestComponent)
    const wrapper = shallow(<BlueClickComponent />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('when the component is clicked', () => {
    it('passes the on prop with true', () => {
      const TestComponent = () => <div>test</div>
      const BlueClickComponent = withBlueClickHOC(TestComponent)
      const wrapper = shallow(<BlueClickComponent />)
      wrapper.simulate('click')

      expect(wrapper.find('TestComponent').prop('on')).toBeTruthy()
    })
  })
})
