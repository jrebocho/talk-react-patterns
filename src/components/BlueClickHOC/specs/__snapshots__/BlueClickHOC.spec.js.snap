// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.withBlueClickHOC renders correctly 1`] = `
<div
  className="blue-box"
  onClick={[Function]}
>
  <TestComponent
    on={false}
  />
</div>
`;
